### PR TITLE
Adding support for Twitter API V2

### PR DESCRIPTION
**Pull request by wszeborowskimateusz**
I would treat it as an alpha support. Much has changed in V2 version of Twitter API.

### CODE REVIEW

File: lib/models/api/v2/tweet_v2.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import 'package:tweet_ui/models/api/v2/entities/tweet_entities_v2.dart';
+import 'package:tweet_ui/models/api/v2/user_v2.dart';
+
+part 'tweet_v2.g.dart';
+
+/// Comparing this model to Twitter API V1, it is important to notice that in V2 most of the fields are optional.
+/// The only required fields are id and text. But this package might require some more fields to properly display tweets.
+/// Those need to be included in expansion parameters.
+@JsonSerializable()
+class TweetV2Response {
```
Comment (kawa89) [inline lib/models/api/v2/tweet_v2.dart:14]: Class name does not correspond to file name.
Maybe we can go with `TweerV2` ?

**Review by kawa89 — CHANGES_REQUESTED** [review]

File: lib/models/api/v2/tweet_v2.dart (hunk 1 above)
Comment (wszeborowskimateusz) [reply lib/models/api/v2/tweet_v2.dart:14]: There is a class TweetV2 in the file. Naming file this way I wanted to stay consistent with API V1.1

**Comment by kawa89** [comment]
Great work! Wow!

1. In Tweet V2 videos are not supported (check links below), so you need to remove the support for that (also remove the fallback video URL) and add an `IMPORTANT` note to the Chengelog about it.
https://twittercommunity.com/t/where-would-i-find-the-direct-link-to-an-mp4-video-posted-in-v2/146933/2
https://twitterdevfeedback.uservoice.com/forums/930250-twitter-api/suggestions/41291761-media-fields-should-return-url-for-gifs-or-videos

2. Delete: 
  factory TweetV2.fromRawJson(String str) => TweetV2.fromJson(json.decode(str));

3. Rename `Tweet` to `TweetV1Response` and add a  `BREAKING` information in the Changelog. Remember to rename all related code that mentioned a Tweet in the v1 version to explicit say that it is for v1 (for example: `EmbeddedTweetView.fromTweet` should be `EmbeddedTweetView.fromTweetV1`).

4. Add missing files to the barrel file (tweet_v1 and tweet_v2).

**Comment by kawa89** [comment]
> Here I am not sure but I assume that photos (also media) might this url field. Also this is not entirely true that videos are not supported. All the information is returned except for video url. That is why I thought allowing to pass just the missing info could be valuable option. But if you feel like we should remove it I can do it (not entirely convinced right now tbh).

I was talking about the fallback video URL that is passed to the `TweetV2Response.fromRawJson`. I think we can remove it.

```dart
TweetV2Response.fromRawJson(
            snapshot.data,
          ),
          darkMode: false,
          createdDateDisplayFormat: DateFormat("EEE, MMM d, ''yy"),
          videoUrl: videoUrl, // I was referring to this field.
```

**Comment by wszeborowskimateusz** [comment]
> > Here I am not sure but I assume that photos (also media) might this url field. Also this is not entirely true that videos are not supported. All the information is returned except for video url. That is why I thought allowing to pass just the missing info could be valuable option. But if you feel like we should remove it I can do it (not entirely convinced right now tbh).
> 
> I was talking about the fallback video URL that is passed to the `TweetV2Response.fromRawJson`. I think we can remove it.
> 
> ```dart
> TweetV2Response.fromRawJson(
>             snapshot.data,
>           ),
>           darkMode: false,
>           createdDateDisplayFormat: DateFormat("EEE, MMM d, ''yy"),
>           videoUrl: videoUrl, // I was referring to this field.
> ```

Yeah, I know about it - but still my point is that if we remove it and tweet has video in it, then it will be displayed as errror with no option to fix. And when having this param there is at least an option to supply it outside of standard dto

**Comment by kawa89** [comment]
The V2 version does not support video for now. When they will add it to the response, or provide a way to get the URL then we will think how to deal with it ;)

For now, let us be 1-1 with what Twitter is giving us :)

**Comment by wszeborowskimateusz** [comment]
I can add a description about this fallback value and the whole video situation to a README

**Comment by wszeborowskimateusz** [comment]
Added such readme entry @kawa89

**Comment by wszeborowskimateusz** [comment]
I am not really convinced, but I changed it as you requested

**Review by kawa89 — APPROVED** [review]

**Pull request merged**